### ISSUE TITLE
Error when mpirun -np 3?

### ISSUE THREAD

**Issue by CHNxindong** [Wangt-CN/DisCo#79]
@Wangt-CN @yhZhai @kevinlin311tw Hello, authors. Thanks for your great work! I have a problem: 
When I use 1 gpu, 2 gpus, 4 gpus and 8gpus, the code can run normaly. 
But when I use 3 gpus or 5 gpus, the code cannot run and print following error message: 
![image](https://github.com/Wangt-CN/DisCo/assets/20262070/0e62eafb-dcea-4f4b-a131-85bc2c3baa75)

The command I use is as follows (Note that the error occurs when I only modify the number of gpu):
AZFUSE_USE_FUSE=0 NCCL_ASYNC_ERROR_HANDLING=0 mpirun -np 3 python /mnt/disk_1/DisCo-main/finetune_sdm_yaml.py 
--cf /mnt/disk_1/DisCo-main/config/ref_attn_clip_combine_controlnet/tiktok_S256L16_xformers_tsv.py 
--do_train 
--root_dir /mnt/disk_1/DisCo-main/run_test 
--local_train_batch_size 24 
--local_eval_batch_size 24 
--log_dir /mnt/disk_1/DisCo-main/exp/tiktok_ft 
--epochs 20 
--deepspeed 
--eval_step 500 
--save_step 500 
--gradient_accumulate_steps 1 
--learning_rate 2e-4 
--fix_dist_seed 
--loss_target "noise" 
--train_yaml /mnt/disk_1/DisCo-main/tiktok_data_include_more/TikTok_finetuning/composite_offset/train_TiktokDance-poses-masks.yaml 
--val_yaml /mnt/disk_1/DisCo-main/tiktok_data_include_more/TikTok_finetuning/composite_offset/new10val_TiktokDance-poses-masks.yaml 
--unet_unfreeze_type "all" 
--refer_sdvae 
--ref_null_caption False 
--combine_clip_local 
--combine_use_mask 
--conds "poses" "masks" 
--stage1_pretrain_path /mnt/disk_1/DisCo-main/DisCo_checkpoint/stage1_HAP/1.2m_pretrain/mp_rank_00_model_states.pt 
--drop_ref 0.05 
--guidance_scale 1.5

Could you please give some suggestions? I want to know wheather the hyperparam of gpu number influence the code running? 
Thank you!

**Comment by CHNxindong** [comment]
Update: 
I remove the arg '-deepspeed' in startup command and the code can run with any number of gpu. But I do not know the reason. Hope someone can share some points. 

PS: the verson of deepspeed I installed is the same as readme.md (0.6.3).

**Comment by CHNxindong** [comment]
I solve the problem: reinstall the environment.
